### PR TITLE
#160318853 Create view articles 

### PR DESCRIPTION
**Pull request by johnmusiu**
#### What does this PR do?
Implements view articles feature.

#### Description of tasks to be completed
Implements loading of all articles when a user navigates to the home page.

#### How should this be manually tested?
```sh
$ git clone https://github.com/andela/ah-titans-frontend.git
```
```sh
$ git checkout ft-create-view-articles-160318853
```
```sh
$ cd ah-titans-frontend
```

> Install the dependencies with the command below.

```sh
$ npm install
```
>Then run the app with the command below.
```sh
$ npm start
```
> You should get to the home page (`http://localhost:4000/`)

image

> After the articles have loaded

image

#### Any background context you want to provide?
No

#### What are the relevant Pivotal Tracker Stories?
[#160318853](https://www.pivotaltracker.com/story/show/160318853)

### CODE REVIEW

**Comment by dcDalin** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  :exclamation: Big PR
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/Loader/loader.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// import ArticlesForm from './components/index';
+// import SearchComponent from './components/search';
+
+describe('Loader', () => {
+  it('renders without crashing', () => {
+    shallow(<Loader store={store} />);
+  });
+});
+
+// describe('<SearchComponent />', () => {
```
Comment (evansmusomi) [inline src/Loader/loader.test.js]: Lines 15 - 44 are commented out. Please remove them if they're not adding to the tests.

**Comment by jmnyarega** [comment]
Please fix these conflicts

**Comment by jmnyarega** [comment]
Also consider adding more tests and fixing issues on codeclimate

File: src/Home/components/index.scss [hunk 2]
```diff
@@ -1,7 +1,61 @@
...
+  border: 0.5px solid white;
+  border-radius: 5px;
+  color: green;
+  margin-left: 5px;
+  margin-right: 175px;
+  padding: 1px 5px 1px 5px;
+}
+.readmore {
   color: green;
-}
```
Comment (migot01) [inline src/Home/components/index.scss:7]: Remove the the extra line here

**Review by petermwash — APPROVED** [review]

File: src/Home/components/btnCreateArticle.test.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import CreateArticleButton from './btnCreateArticle';
+
+describe('CreateArticleButton', () => {
+	it('renders without crashing', () => {
+		shallow(<CreateArticleButton />);
```
Comment (jmnyarega) [inline src/Home/components/btnCreateArticle.test.js]: This indentation seems off

File: src/Home/homepage.test.js [hunk 4]
```diff
@@ -1,10 +1,133 @@
...
+				'',
+			tagList: [
+				'any',
+			],
+			body:
+				'this is a test body',
+			created_at:
+				'',
+		},
+	};
```
Comment (jmnyarega) [inline src/Home/homepage.test.js]: Let's restructure this
```
const article = {
  article: {
    slug: "",
    tagList: [ "any",  ],
    body: "This is a test body",
    created_at: "",
  },
};

File: src/Home/homepage.test.js [hunk 5]
```diff
@@ -1,10 +1,133 @@
...
+	it('renders without crashing', () => {
+		shallow(
+			<ArticlesPage
+				store={
+					store
+				}
+				article={
+					article
+				}
+			/>,
```
Comment (jmnyarega) [inline src/Home/homepage.test.js]: Also this needs restructuring
```
  <ArticlesPage
     store={store}
     article={article}
  />

File: src/Home/homepage.test.js [hunk 6]
```diff
@@ -1,10 +1,133 @@
...
+	it('returns 2 tags', () => {
+		expect(
+			articlesFormWrapper.find(
+				'.chip',
+			)
+				.length,
+		).toEqual(
+			0,
+		);
+	});
```
Comment (jmnyarega) [inline src/Home/homepage.test.js]: Kindly restructure all these ⬆️

File: src/actions/createArticlesAction.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      );
+      history.push('/');
+    })
+    .catch(error => dispatch(createArticleError(error)));
+};
+
+createArticle.propTypes = {
+  createArticleActionCreator: PropTypes.func.isRequired,
+  createArticleError: PropTypes.func.isRequired,
+};
```
Comment (jmnyarega) [inline src/actions/createArticlesAction.js]: **WE DON'T NEED proptypes on ACTIONS**

Comment (janetnim) [reply src/actions/createArticlesAction.js]: It was on the feature already merged for Creating an article

**Review by ronaldndirangu — APPROVED** [review]

**Review by migot01 — APPROVED** [review]

**Pull request merged**